eslint.config.js

```diff
@@ -1,15 +1,16 @@
 // @ts-check
 import eslint from "@eslint/js";
-import deprecation from "eslint-plugin-deprecation";
 import reactHooks from "eslint-plugin-react-hooks";
 import unicorn from "eslint-plugin-unicorn";
 import vitest from "eslint-plugin-vitest";
-import { dirname } from "path";
 import tsEslint from "typescript-eslint";
-import { fileURLToPath } from "url";
 
 /** Config that will apply to all files */
 const allFilesConfig = tsEslint.config({
+  linterOptions: {
+    // This is problematic for no-deprecated rule which needs dependencies to be built to generate the d.ts files and know that it is deprecated
+    // reportUnusedDisableDirectives: "off",
```

Comment:
Did you work around this? Looks like you're not turning off this flag now.

Reply:
Edit: Oh I see you remove deprecation plugin later

Reply:
ah yes, using the new projectService from eslint allows it to load the source as long as the tsconfig have `references`